app/Http/Controllers/Api/PublisherController.php

```diff
@@ -31,11 +34,58 @@ public function createPublisher(Request $request)
             }
 
             return response()->json($publisher);
-        } catch (\Exception $e) {
+        } catch (Exception $e) {
             return response()->json(['error' => 'An error occurred while trying to create the publisher'], 500);
         }
     }
 
+    public function viewAllPublishers(Request $request)
+    {
+        try {
+            $perPageOptions = [20, 50, 100];
+
+            $searchValue = $request->input('search', '');
+            $perPage = $request->input('per_page', 20);
+
+            if (!in_array($perPage, $perPageOptions)) {
+                $perPage = 20;
+            }
+
+            $validator = validator()->make($request->all(), [
+                'search' => 'string',
+                'per_page' => 'integer|in:' . implode(',', $perPageOptions),
+            ]);
+
+            if ($validator->fails()) {
+                return response()->json([
+                    'error' => 'Invalid parameters',
+                    'message' => $validator->errors()
+                ], 400);
+            }
+
+            $query = Publisher::query();
+            if (!empty($searchValue)) {
```

Comment:
@damir-tahirovic This is fine, but you can use Laravel's when method here instead for better readability, like so:
```php
$query = Publisher::query()
            ->when($searchValue, function ($q) use ($searchValue) {
                $q->where('name', 'LIKE', "%{$searchValue}%");
            });
```